### PR TITLE
Transform custom PyEvent into standard python dictionary for easier d…

### PR DESCRIPTION
**Pull request by haixuanTao**
…ebuggability.

Currently having a custom PyEvent make debugging very hard as fields are hidden within the class PyEvent that is defined within Rust Code.

Python user are getting really confused about this obscure class.

This PR transforms the class into a standard python dictionary.

It also makes external event and dora event expose the same user API meaning to access the data people just use:

`data = event["value"]` instead of `data = event.inner()`

### CODE REVIEW

File: apis/python/node/src/lib.rs [hunk 1]
```diff
@@ -84,10 +91,11 @@ impl Node {
...
     /// ```
     ///
+    /// Default behaviour is to timeout after 2 seconds.
+    ///
     /// :rtype: dora.PyEvent
-    pub fn __next__(&mut self, py: Python) -> PyResult<Option<PyEvent>> {
-        let event = py.allow_threads(|| self.events.recv(None));
-        Ok(event)
+    pub fn __next__(&mut self, py: Python) -> PyResult<Option<Py<PyDict>>> {
+        self.next(py, Some(2.0))
```
Comment (phil-opp) [inline apis/python/node/src/lib.rs]: This default timeout change does not seem related to this PR. If that's intended, could you open a separate PR for it so that it is more visible?

File: apis/python/operator/src/lib.rs [hunk 2]
```diff
@@ -1,69 +1,53 @@
...
-    /// :rtype: dora.PyObject
-    pub fn __getitem__(&self, key: &str, py: Python<'_>) -> PyResult<Option<PyObject>> {
-        if key == "kind" {
-            let kind = match &self.event {
-                MergedEvent::Dora(_) => "dora",
-                MergedEvent::External(_) => "external",
-            };
-            return Ok(Some(kind.to_object(py)));
-        }
+    pub fn to_py_dict_bound(self, py: Python<'_>) -> PyResult<Py<PyDict>> {
```
Comment (phil-opp) [inline apis/python/operator/src/lib.rs:19]: I think we can remove the `_bound` part from the function name since it doesn't return a `Bound` type.

File: examples/python-dataflow/requirements.txt [hunk 3]
```diff
@@ -6,7 +6,7 @@ ultralytics
 gitpython
 ipython  # interactive notebook
 matplotlib>=3.2.2
-numpy>=1.18.5
+numpy>=1.18.5,<2.0.0
```
Comment (phil-opp) [inline examples/python-dataflow/requirements.txt:9]: This is caused by the numpy 2 release, right? I'm also seeing the same CI errors for #558 .

**Review by phil-opp — APPROVED** [review]
Looks good overall!

Is this a breaking change or does existing code continue to work?

**Comment by haixuanTao** [comment]
Ezisting code will continue to work!

Except for ros2-bridge where you will have to change `data.inner()` -> `data["value"]` but I think that it is still okay as a breaking change

File: apis/python/operator/src/lib.rs (hunk 2 above)
Comment (haixuanTao) [reply apis/python/operator/src/lib.rs:19]: Indeed

File: examples/python-dataflow/requirements.txt (hunk 3 above)
Comment (haixuanTao) [reply examples/python-dataflow/requirements.txt:9]: Yes! I'm going to push a fix on this branch

File: apis/python/node/src/lib.rs (hunk 1 above)
Comment (haixuanTao) [reply apis/python/node/src/lib.rs]: Okay sure!

Comment (haixuanTao) [reply apis/python/node/src/lib.rs]: I will leave it to None in that case on this PR

File: examples/python-dataflow/requirements.txt (hunk 3 above)
Comment (phil-opp) [reply examples/python-dataflow/requirements.txt:9]: Thanks a lot!

Looks like it's related to `cv2`: https://github.com/opencv/opencv-python/issues/997

**Comment by phil-opp** [comment]
> Ezisting code will continue to work!
> 
> Except for ros2-bridge where you will have to change `data.inner()` -> `data["value"]` but I think that it is still okay as a breaking change

Ok great! The ROS2 bridge change is fine I think given that we still consider it experimental.

**Comment by haixuanTao** [comment]
> > Ezisting code will continue to work!
> > Except for ros2-bridge where you will have to change `data.inner()` -> `data["value"]` but I think that it is still okay as a breaking change
> 
> Ok great! The ROS2 bridge change is fine I think given that we still consider it experimental.

Yup. It makes the current implementation of the ros2-bridge closer to the rest of dora API
